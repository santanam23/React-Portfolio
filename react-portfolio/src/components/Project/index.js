import React from 'react';
import foodFest from '../../assets/images/foodFest.png';
import notetaker from '../../assets/images/notetaker.png';
import passwordGen from '../../assets/images/passwordGen.png';
import weatherDash from '../../assets/images/weatherDash.png';

function Project() {
    return (
    <section className="mx-5" id="project">
        <h2>Projects</h2>
            <div className="gallery">
            <a target="_blank" href="foodFest.pgn">
                <img src={foodFest} alt="Food Festival" width="600" height="400" />
            </a>
            <div className="desc">This is a link to my Food Fest PWA</div>
            </div>

            <div className="gallery">
            <a target="_blank" href="notetaker.pgn">
                <img src={notetaker} alt="Notetaker" width="600" height="400" /> 
            </a>
            <div className="desc">This is a link to Notetaker app</div>
            </div>

            <div className="gallery">
            <a target="_blank" href="passwordGen.png">
                <img src={passwordGen} alt="Password Generator" width="600" height="400" />
            </a>
            <div className="desc">This is a link to Password Generator website</div>
            </div>

            <div className="gallery">
            <a target="_blank" href="weatherDash.png">
                <img src={weatherDash} alt="Weather Dashboard" width="600" height="400" />
            </a>
            <div className="desc">This is a link to the Weather Dashboard</div>
            </div>

    </section>
    )
}
export default Project;