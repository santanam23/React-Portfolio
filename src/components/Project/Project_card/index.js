import * as React from 'react';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';

export default function ProjectCard({name,description,image}) {
  return (
    <Card sx={{ maxWidth: 345 }}>
      <CardMedia
        component="img"
        height="140"
        image={image}
        alt={name}
      />
      <CardContent>
        <Typography gutterBottom variant="h5" component="div">
          {name}
        </Typography>
        <Typography variant="body2" color="text.secondary">
          {description}
        </Typography>
      </CardContent>
      <CardActions>
        <Button size="small">Link</Button>
        <Button size="small">Learn More</Button>
      </CardActions>
    </Card>
  );
}
