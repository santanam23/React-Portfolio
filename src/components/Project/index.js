import React from 'react';
import foodFest from '../../assets/images/foodFest.png';
import notetaker from '../../assets/images/notetaker.png';
import passwordGen from '../../assets/images/passwordGen.png';
import weatherDash from '../../assets/images/weatherDash.png';
import houstonVolleyballSocial from '../../assets/images/houstonVolleyballSocial.png';

function Project() {
    return (
    <section className="mx-5" id="project">
        <h2>Projects</h2>
            <div className="gallery">
            <a target="_blank" rel="noopener noreferrer" href="https://santanam23.github.io/food-festival/">
                <img src={foodFest} alt="Food Festival" width="600" height="400" />
            </a>
            <div className="desc">This is a link to my Food Fest PWA</div>
            </div>

            <div className="gallery">
            <a target="_blank" rel="noopener noreferrer" href="https://gentle-atoll-41675.herokuapp.com/">
                <img src={notetaker} alt="Notetaker" width="600" height="400" /> 
            </a>
            <div className="desc">This is a link to Notetaker app</div>
            </div>

            <div className="gallery">
            <a target="_blank" rel="noopener noreferrer" href="https://santanam23.github.io/Password-Generator/">
                <img src={passwordGen} alt="Password Generator" width="600" height="400" />
            </a>
            <div className="desc">This is a link to Password Generator website</div>
            </div>

            <div className="gallery">
            <a target="_blank" rel="noopener noreferrer" href="https://santanam23.github.io/Weather-Dashboard/">
                <img src={weatherDash} alt="Weather Dashboard" width="600" height="400" />
            </a>
            <div className="desc">This is a link to the Weather Dashboard</div>
            </div>

            <div className="gallery">
            <a target="_blank" rel="noopener noreferrer" href="https://secret-beyond-82045.herokuapp.com/">
                <img src={houstonVolleyballSocial} alt="Houston Volleyball Social" width="600" height="400" /> 
            </a>
            <div className="desc">This is a link to my Project</div>
            </div>
    </section>
    )
}
export default Project;