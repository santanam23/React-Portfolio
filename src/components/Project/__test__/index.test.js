import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Project from "..";

afterEach(cleanup)

describe('Project component', () => {
    // First Test
    it('renders', () => {
      render(<Project />);
    })
    it('matches snapshot DOM node structure', () => {
      // render Project
      const { asFragment } = render(<Project />);
      expect(asFragment()).toMatchSnapshot();
    })
});