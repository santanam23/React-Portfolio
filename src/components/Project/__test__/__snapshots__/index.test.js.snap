// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="mx-5"
    id="project"
  >
    <h2>
      Projects
    </h2>
    <div
      class="gallery"
    >
      <a
        href="foodFest.pgn"
        target="_blank"
      >
        <img
          alt="Food Festival"
          height="400"
          src="foodFest.png"
          width="600"
        />
      </a>
      <div
        class="desc"
      >
        This is a link to my Food Fest PWA
      </div>
    </div>
    <div
      class="gallery"
    >
      <a
        href="notetaker.pgn"
        target="_blank"
      >
        <img
          alt="Notetaker"
          height="400"
          src="notetaker.png"
          width="600"
        />
      </a>
      <div
        class="desc"
      >
        This is a link to Notetaker app
      </div>
    </div>
    <div
      class="gallery"
    >
      <a
        href="passwordGen.png"
        target="_blank"
      >
        <img
          alt="Password Generator"
          height="400"
          src="passwordGen.png"
          width="600"
        />
      </a>
      <div
        class="desc"
      >
        This is a link to Password Generator website
      </div>
    </div>
    <div
      class="gallery"
    >
      <a
        href="weatherDash.png"
        target="_blank"
      >
        <img
          alt="Weather Dashboard"
          height="400"
          src="weatherDash.png"
          width="600"
        />
      </a>
      <div
        class="desc"
      >
        This is a link to the Weather Dashboard
      </div>
    </div>
  </section>
</DocumentFragment>
`;
